pandas/util/_decorators.py

```diff
@@ -290,14 +290,29 @@ def deprecate_nonkeyword_arguments(
     """
 
     def decorate(func):
+        old_sig = inspect.signature(func)
+
         if allowed_args is not None:
             allow_args = allowed_args
         else:
-            spec = inspect.getfullargspec(func)
+            allow_args = [
+                p.name
+                for p in old_sig.parameters.values()
+                if p.kind in (p.POSITIONAL_ONLY, p.POSITIONAL_OR_KEYWORD)
```

Comment:
Is there a test that requires `p.POSITIONAL_ONLY`? I tried deleting it from here, and `pytest pandas/tests/util/test_deprecate_nonkeyword_arguments.py` still passed

If there's not a test that needs this, could we add a little one that does?

EDIT: apologies, I wrote this comment then forgot to add this comment to the previous review

Reply:
Added a test!